### PR TITLE
If only a custom Ressource provider is provided old version failed --> now fixed

### PR DESCRIPTION
**Pull request by BenjaminArp**

### CODE REVIEW

**Review by 01Parzival10 — APPROVED** [review]
LGTM

**Comment by 01Parzival10** [comment]
Wait before merging. I'm currently writing a test or two for that

**Review by 01Parzival10 — DISMISSED** [review]
Can be merged after @Entenwilli had a look at it

File: tests/org.dataflowanalysis.analysis.tests/src/org/dataflowanalysis/analysis/tests/dfd/ResourceProviderTest.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        analysis.initializeAnalysis();
+        
+        var flowGraphs = analysis.findFlowGraphs();
+		flowGraphs.evaluate();
+		
+		assertEquals(flowGraphs.getTransposeFlowGraphs()
+                .size(), 4);
+	}
+	
+	@SuppressWarnings({ "rawtypes", "unchecked" })
```
Comment (Entenwilli) [inline tests/org.dataflowanalysis.analysis.tests/src/org/dataflowanalysis/analysis/tests/dfd/ResourceProviderTest.java:114]: Could we get rid of this suppress warnings? It is probably caused by the usage of a class that has generics

File: tests/org.dataflowanalysis.analysis.tests/src/org/dataflowanalysis/analysis/tests/dfd/ResourceProviderTest.java [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+	     for (String fileext : fileextensions) {
+	        rs.getResourceFactoryRegistry().getExtensionToFactoryMap().put(fileext, new XMLResourceFactoryImpl());
+	     }		
+	     URI uri = URI.createFileURI(outputFile);
+	     Resource resource = rs.createResource(uri);
+	     ((ResourceImpl)resource).setIntrinsicIDToEObjectMap(new HashMap());
+	     return resource;
+	  }
+	
+	@SuppressWarnings({"unchecked", "rawtypes"})
```
Comment (Entenwilli) [inline tests/org.dataflowanalysis.analysis.tests/src/org/dataflowanalysis/analysis/tests/dfd/ResourceProviderTest.java:125]: Could we get rid of this suppress warnings? It is probably caused by the usage of a class that has generics

**Review by Entenwilli — CHANGES_REQUESTED** [review]
Only one smaller comment about the `SuppressWarnings` annotations

File: tests/org.dataflowanalysis.analysis.tests/src/org/dataflowanalysis/analysis/tests/dfd/ResourceProviderTest.java (hunk 1 above)
Comment (01Parzival10) [reply tests/org.dataflowanalysis.analysis.tests/src/org/dataflowanalysis/analysis/tests/dfd/ResourceProviderTest.java]: Done

**Pull request merged**